### DISCUSSION TITLE
so-checkin fails: Specified SLS 'data.*' in environment 'base' is not available on the salt master

### DISCUSSION THREAD

**Discussion by DebianGuru** [Security-Onion-Solutions/securityonion#12605] [2.4]
### Version

2.4.50

### Installation Method

Security Onion ISO image

### Description

installation

### Installation Type

Standalone

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

12

### RAM

20gm

### Storage for /

200g

### Storage for /nsm

1.5TB

### Network Traffic Collection

span port

### Network Traffic Speeds

Less than 1Gbps

### Status

No, one or more services are failed (please provide detail below)

### Salt Status

Yes, there are salt failures (please provide detail below)

### Logs

Yes, there are additional clues in /opt/so/log/ (please provide detail below)

### Detail

I have no idea how this happened, but salt is refusing to start. The install was not working before.  I could see nothing from zeek or suricata.  After lots of troublehooting, I see this on "so-checkin"
`# so-checkin
[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/grains'
[INFO    ] Syncing grains for environment 'base'
[INFO    ] Loading cache from salt://_grains, for base
[INFO    ] Caching directory '_grains/' for environment 'base'
[INFO    ] Loading fresh modules for state activity
local:
    Data failed to compile:
----------
    Pillar failed to render with the following messages:
----------
    Specified SLS 'data.*' in environment 'base' is not available on the salt master
----------
    Specified SLS 'global' in environment 'base' is not available on the salt master
`

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by cm-ops** [comment]
A couple of questions. Did the install complete successfully? What returns if you run `sudo salt-call state.show_top`?

**Reply by DebianGuru** [reply]
@cm-ops Thanks for the reply.

Heres what is returned from that command:
[INFO    ] Creating module dir '/var/cache/salt/minion/extmods/grains'
[INFO    ] Syncing grains for environment 'base'
[INFO    ] Loading cache from salt://_grains, for base
[INFO    ] Caching directory '_grains/' for environment 'base'
[INFO    ] Loading fresh modules for state activity
local:
    ----------
    base:
        - cron.running
        - repo.client
        - salt.minion-state-apply-test
        - salt.minion

**Reply by DebianGuru** [reply]
Also, I don't know if this will help, but so-checkin returns:
.
.
.
[INFO    ] Executing command rpm in directory '/root'
local:
    Data failed to compile:
----------
    Specified SLS repo.client.oel in saltenv base is not available on the salt master or through a configured fileserver

I asume it's looking for some YUM repo, but I'm not sure how to fix that.

**Reply by cm-ops** [reply]
Looks like the top file is missing a few things. What do you see if you run `ls /opt/so/saltstack/default/salt`? Is there a backup `top.sls`? If not, compare yours against what it should be - https://github.com/Security-Onion-Solutions/securityonion/blob/2.4/main/salt/top.sls

If it is not what it should be, replace with the correct configuration and retry `so-checkin`.
